tools/rebuild.sh

```diff
@@ -60,6 +60,14 @@ function try() {
     "$@" || die "test failed: $*"
 }
 
+check_error()
+{
+    if [ "$?" = "1" ]; then
+        echo "Directory doesn't exist!" 1>&2
+        exit 1
+    fi
+}
+
```

Comment:
Good idea--it would make diagnosing problems easier.

This could be simpler if we put both `cd` and the error check into 1 function. Then instead of:
```
cd $TCF_HOME/common/sgx_workload/
check_error
```
You only need:
`cd_check $TCF_HOME/common/sgx_workload/`

The check should check the `$?` status to not be 0--in case cd has other exit codes besides 0 or 1.
Also generalize the message.  The user may not have execute (`x`) permission to `cd` to the directory.

The `cd_check` function could be something like:
```
cd_check()
{
    if [ ! -d "$1" ] ; then
        echo "Directory does not exist: $1" 1>&2
        exit 1
    fi

   cd $1
    if [ "$?" != "0" ]; then
        echo "Cannot change to directory $1" 1>&2
        exit 1
    fi
}
```

Reply:
Done

Reply:
How about `cd $TCF_HOME/common/sgx_workload || panic "Failed to change to the directory"`

Where

```
function panic()
{
    echo "[Error]: " ${*}
    exit 1
}
```

The error "No such file or directory" would be printed already. "Cannot change to directory" can occur for permissions which is not caught in the current code anyways.

Reply:
Looks good. You could rename panic to something like error_exit as the function is not existing with error code instead of panic. Also we can invoke with LINE NUM to help in debugging something like `cd $TCF_HOME/common/sgx_workload || error_exit "$LINENO: Failed to change to the directory"`

Reply:
error_exit is close to the indended change. No need to introduce $LINENO. Line No is already printed in the console when error occurs.